### DISCUSSION TITLE
Running update from cron and not overwriting local changes?

### DISCUSSION THREAD

**Discussion by adamshand** [twpayne/chezmoi#3513] [Q&A] [answered]
Hello,

I've been using chezmoi for a while, really makes life easier ... thank you!

I'd like to run `chezmoi update` nightly on all my servers to make sure they always have the latest version of files.  However if there are local changes I'd like them to be skipped, so I don't risk losing something useful.

This seems like it should be easy, but I can't figure out how to do this.  I assume I'm missing something obvious? Any pointers would be appreciated.

Thanks again,
Adam.

**Comment by twpayne** [comment]
You can use the first character of each column of the output of [`chezmoi status`](https://www.chezmoi.io/reference/commands/status/). If any of these characters are not a space then that entry has been modified since chezmoi last wrote it.

**Reply by adamshand** [reply]
Thanks for the reply, but sorry I don't understand.

When I run `chezmoi update` interactively, it prompts me if a local file has been edited and asks if I want to skip updating the local file from github or overwrite the local file from github.

I'm wanting to run `chezmoi update` from cron and have it skip any locally updated files.

How does the first column of `chezmoi status` help?

**Reply by twpayne** [reply, marked as answer]
You can write a script that calls `chezmoi status` and then calls `chezmoi apply` if there are no columns with a non-space in the first column.

**Reply by adamshand** [reply]
Got it, thank you!

In case it's helpful to anyone else, here's a (lightly tested) script for updating dotfiles without overwriting any files with local changes.

```bash
#!/bin/bash

echo "## Pulling latest from GitHub"
chezmoi git pull

# get a list of files to update that don't have local changes
FILES=$(chezmoi status | awk '/^ / {print $2}')

echo -e "\n## Updating files without local changes"

for f in $FILES; do
  echo "... $f"
  chezmoi apply ~/${f}
done

echo -e "\n## Files with local changes that need adding"
chezmoi status 1>&2
```
